pkg/machinery/config/types/v1alpha1/generate/generate.go

```diff
@@ -439,14 +445,26 @@ func NewInput(clustername, endpoint, kubernetesVersion string, secrets *SecretsB
 
 	additionalSubjectAltNames = append(additionalSubjectAltNames, options.AdditionalSubjectAltNames...)
 
+	clusterID, err := genV4UUID()
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate ClusterID: %w", err)
+	}
+
+	clusterSecret, err := nethelpers.GenerateWireguardKey()
```

Comment:
should we tie cluster secret to the wireguard key directly? or should it be more of random X bytes which we can use to derive wireguard key? (and potentially other keys if we need to?)

Reply:
Yeah, this is merely shorthand.  Ultimately, `GenerateWireguardKey` just generates 32 bytes of random data.  We can absolutely use a larger key and trim as needed to match the 32-byte requirement for Wireguard pre-shared keys.

Reply:
I kept it 32 bytes, but I'm using a local random bytes generator to avoid confusion with Wireguard.